prober/derp.go

```diff
@@ -418,6 +418,23 @@ func runDerpProbeQueuingDelayContinously(ctx context.Context, from, to *tailcfg.
 	txRecords := make([]txRecord, 0, packetsPerSecond*int(packetTimeout.Seconds()))
 	var txRecordsMu sync.Mutex
 
+	// applyTimeouts walks over txRecords and expires any records that are older
+	// than packetTimeout, recording in metrics that they were removed.
+	applyTimeouts := func() {
+		txRecordsMu.Lock()
+		defer txRecordsMu.Unlock()
+
+		recs := txRecords[:0]
+		for _, r := range txRecords {
+			if time.Since(r.at) > packetTimeout {
```

Comment:
:question: We aren’t worried about time.Now being slow on a kernel that doesn’t have VDSO for gettimeofday, right?

Reply:
Fair question.

Are we running on anything that old?

There is an issue with the very old Xen VMs where it was still actually a hypercall too!

Will fix